File: android-client/app/src/main/java/com/example/stripedemo/shared/components/AccountSelector.kt
package com.example.stripedemo.shared.components

import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.ArrowDropDown
import androidx.compose.material.icons.filled.Check
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.example.stripedemo.data.models.Account

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun AccountSelector(
    accounts: List<Account>,
    selectedAccount: Account?,
    onAccountSelected: (Account) -> Unit,
    onCreateNewAccount: () -> Unit,
    modifier: Modifier = Modifier
) {
    var expanded by remember { mutableStateOf(false) }

    ExposedDropdownMenuBox(
        expanded = expanded,
        onExpandedChange = { expanded = it },
        modifier = modifier
    ) {
        OutlinedTextField(
            value = selectedAccount?.let { "${it.displayName ?: it.email ?: "Unknown"}" } ?: "Select an account",
            onValueChange = {},
            readOnly = true,
            label = { Text("Account") },
            trailingIcon = {
                Icon(
                    imageVector = Icons.Default.ArrowDropDown,
                    contentDescription = "Dropdown"
                )
            },
            modifier = Modifier
                .fillMaxWidth()
                .menuAnchor()
        )

        ExposedDropdownMenu(
            expanded = expanded,
            onDismissRequest = { expanded = false }
        ) {
            accounts.forEach { account ->
                DropdownMenuItem(
                    text = {
                        Row(
                            modifier = Modifier.fillMaxWidth(),
                            horizontalArrangement = Arrangement.SpaceBetween,
                            verticalAlignment = Alignment.CenterVertically
                        ) {
                            Column {
                                Text(account.displayName ?: account.email ?: "Unknown")
                                Text(
                                    text = if (account.isRecipient) "Customer + Recipient" else "Customer",
                                    style = MaterialTheme.typography.bodySmall,
                                    color = MaterialTheme.colorScheme.onSurfaceVariant
                                )
                            }
                            if (selectedAccount?.id == account.id) {
                                Icon(
                                    imageVector = Icons.Default.Check,
                                    contentDescription = "Selected",
                                    tint = MaterialTheme.colorScheme.primary
                                )
                            }
                        }
                    },
                    onClick = {
                        onAccountSelected(account)
                        expanded = false
                    }
                )
            }

            HorizontalDivider()

            DropdownMenuItem(
                text = {
                    Text(
                        "+ Create New Account",
                        color = MaterialTheme.colorScheme.primary
                    )
                },
                onClick = {
                    onCreateNewAccount()
                    expanded = false
                }
            )
        }
    }
}

File: ios-client/StripeApplication/StripeApplication/Shared/Views/AccountSelector.swift
import SwiftUI

struct AccountSelector: View {
    let accounts: [Account]
    let selectedAccount: Account?
    let onAccountSelected: (Account) -> Void
    let onCreateNewAccount: () -> Void
    
    

    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            Text("Account")
                .font(.caption)
                .foregroundStyle(.secondary)

            Menu {
                ForEach(accounts) { account in
                    Button {
                        onAccountSelected(account)
                    } label: {
                        HStack {
                            VStack(alignment: .leading) {
                                Text(account.displayName ?? account.email ?? "Unknown")
                                Text(account.isRecipient ? "Customer + Recipient" : "Customer")
                                    .font(.caption)
                                    .foregroundStyle(.secondary)
                            }
                            if selectedAccount?.id == account.id {
                                Spacer()
                                Image(systemName: "checkmark")
                            }
                        }
                    }
                }

                Divider()

                Button {
                    onCreateNewAccount()
                } label: {
                    Label("Create New Account", systemImage: "plus")
                }
            } label: {
                HStack {
                    Text(selectedAccount.map { $0.displayName ?? $0.email ?? "Unknown" } ?? "Select an account")
                        .foregroundStyle(selectedAccount != nil ? .primary : .secondary)
                    Spacer()
                    Image(systemName: "chevron.up.chevron.down")
                        .foregroundStyle(.secondary)
                }
                .padding()
                .background(
                    RoundedRectangle(cornerRadius: 10)
                        .stroke(Color.secondary.opacity(0.3), lineWidth: 1)
                )
            }
        }
    }
}
